External/Plugins/HaXeContext/Linters/DiagnosticsLinter.cs

```diff
@@ -26,47 +30,75 @@ public void LintAsync(string[] files, LintCallback callback)
+                var i1 = i;
+                
+                hc.GetDiagnostics((complete, results, status) =>
+                {
+                    if (status == HaxeCompleteStatus.DIAGNOSTICS && results != null && sci != null)
                     {
-                        var line = res.Range.LineStart + 1;
-                        var firstChar = sci.PositionFromLine(line) + res.Range.CharacterStart;
-                        var lastChar = sci.PositionFromLine(res.Range.LineEnd + 1) + res.Range.CharacterEnd;
-                        var result = new LintingResult
+                        foreach (var res in results)
                         {
-                            File = res.Range.Path,
-                            FirstChar = res.Range.CharacterStart,
-                            Length = lastChar - firstChar,
-                            Line = line,
-                            Severity = DiagnosticsSeverityToLintingSeverity[res.Severity]
-                        };
+                            var line = res.Range.LineStart + 1;
+                            var firstChar = sci.PositionFromLine(line) + res.Range.CharacterStart;
+                            var lastChar = sci.PositionFromLine(res.Range.LineEnd + 1) + res.Range.CharacterEnd;
+                            var result = new LintingResult
+                            {
+                                File = res.Range.Path,
+                                FirstChar = res.Range.CharacterStart,
+                                Length = lastChar - firstChar,
+                                Line = line,
+                                Severity = DiagnosticsSeverityToLintingSeverity[res.Severity]
+                            };
 
-                        switch (res.Kind)
-                        {
-                            case HaxeDiagnosticsKind.UNUSEDIMPORT:
-                                result.Description = TextHelper.GetString("HaXeContext.Info.UnusedImport");
-                                break;
-                            case HaxeDiagnosticsKind.UNUSEDVAR:
-                                result.Description = TextHelper.GetString("HaXeContext.Info.UnusedVariable");
-                                break;
-                        }
+                            switch (res.Kind)
+                            {
+                                case HaxeDiagnosticsKind.UNUSEDIMPORT:
+                                    result.Description = TextHelper.GetString("HaXeContext.Info.UnusedImport");
+                                    break;
+                                case HaxeDiagnosticsKind.UNUSEDVAR:
+                                    result.Description = TextHelper.GetString("HaXeContext.Info.UnusedVariable");
+                                    break;
+                                case HaxeDiagnosticsKind.UNUSEDCASE:
+                                    result.Description = TextHelper.GetString("HaXeContext.Info.UnusedCase");
```

Comment:
FYI: `"HaXeContext."` is redundant, and slower since `TextHelper.GetString()` will try to look for `"HaXeContext.HaXeContext.Info.UnusedCase"` first.